### PR TITLE
Auth: makes login redirect with error public

### PR DESCRIPTION
**Pull request by xlson**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by aknuds1 — APPROVED** [review]
LGTM - Is this change necessary for use from enterprise?

**Comment by xlson** [comment]
@aknuds1 yeah. I'd like to solve this differently longterm, but I think it makes sense for starters. It makes it possible to redirect with an error from the enterprise saml integration.

File: pkg/api/login_oauth.go [hunk 1]
```diff
@@ -243,7 +243,7 @@ func hashStatecode(code, seed string) string {
 	return hex.EncodeToString(hashBytes[:])
 }
 
-func (hs *HTTPServer) redirectWithError(ctx *models.ReqContext, err error, v ...interface{}) {
+func (hs *HTTPServer) RedirectWithError(ctx *models.ReqContext, err error, v ...interface{}) {
```
Comment (bergquist) [inline pkg/api/login_oauth.go:246]:
```suggestion
// RedirectWithError redirects the requests with an error message.
func (hs *HTTPServer) RedirectWithError(ctx *models.ReqContext, err error, v ...interface{}) {
```

**Comment by xlson** [comment]
Closing this as @alexanderzobnin implemented this in another PR.

**Comment by alexanderzobnin** [comment]
Ref #23391

**Pull request closed without merging**